### PR TITLE
PN-10950-be-api-exchange-qr-code-pf-pg

### PR DESCRIPTION
**Pull request by ImJimMoriarty**
This PR introduces a proxy for the aar_qr_check in selfcare.

### CODE REVIEW

File: src/test/java/it/pagopa/pn/bff/mocks/NotificationsReceivedMock.java [hunk 1]
```diff
@@ -52,4 +54,15 @@ public NotificationSearchResponse getNotificationReceivedPNMock() {
         notificationSearchResponse.setMoreResult(false);
         return notificationSearchResponse;
     }
+    public ResponseCheckAarMandateDto getResponseCheckAarMandateDtoPNMock(){
```
Comment (AndreaCimini90) [inline src/test/java/it/pagopa/pn/bff/mocks/NotificationsReceivedMock.java:57]: add an empty line

File: src/test/java/it/pagopa/pn/bff/mocks/NotificationsReceivedMock.java [hunk 2]
```diff
@@ -52,4 +54,15 @@ public NotificationSearchResponse getNotificationReceivedPNMock() {
...
+        ResponseCheckAarMandateDto responseCheckAarMandateDto = new ResponseCheckAarMandateDto();
+        responseCheckAarMandateDto.setMandateId("MANDATE_ID");
+        responseCheckAarMandateDto.setIun("IUN");
+        return responseCheckAarMandateDto;
+    }
+    public BffRequestCheckAarMandateDto getRequestCheckAarMandateDtoPNMock(){
```
Comment (AndreaCimini90) [inline src/test/java/it/pagopa/pn/bff/mocks/NotificationsReceivedMock.java:63]: add an empty line

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: docs/openapi/api-internal-pn-bff-recipient-notifications.yaml [hunk 3]
```diff
@@ -245,4 +245,37 @@ paths:
...
+      operationId: checkAarQrCodeV1
+#      security:                                      # ONLY EXTERNAL
+#        - bearerAuth: [ ]                            # ONLY EXTERNAL
+      parameters: # NO EXTERNAL
+      - $ref: './common-refs.yaml#/components/parameters/uidAuthFleet'                # NO EXTERNAL
+      - $ref: './common-refs.yaml#/components/parameters/cxTypeAuthFleet'             # NO EXTERNAL
+      - $ref: './common-refs.yaml#/components/parameters/cxIdAuthFleet'               # NO EXTERNAL
+      - $ref: './common-refs.yaml#/components/parameters/cxGroupsAuthFleet'           # NO EXTERNAL
+      requestBody:
+        content:
```
[the remark this answers is not in the document]
Comment (ImJimMoriarty) [reply docs/openapi/api-internal-pn-bff-recipient-notifications.yaml:264]: it is already present below

File: src/main/java/it/pagopa/pn/bff/mappers/notifications/NotificationAarQrCodeMapper.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+import it.pagopa.pn.bff.generated.openapi.msclient.delivery_recipient.model.RequestCheckAarMandateDto;
+import it.pagopa.pn.bff.generated.openapi.msclient.delivery_recipient.model.ResponseCheckAarMandateDto;
+import it.pagopa.pn.bff.generated.openapi.server.v1.dto.BffCheckAarRequest;
+import it.pagopa.pn.bff.generated.openapi.server.v1.dto.BffCheckAarResponse;
+import org.mapstruct.Mapper;
+import org.mapstruct.factory.Mappers;
+
+@Mapper
+public interface NotificationAarQrCodeMapper {
```
Comment (AndreaCimini90) [inline src/main/java/it/pagopa/pn/bff/mappers/notifications/NotificationAarQrCodeMapper.java:11]: Add java doc here

File: src/test/java/it/pagopa/pn/bff/mappers/notifications/NotificationsReceivedMapperTest.java [hunk 5]
```diff
@@ -36,4 +40,31 @@ void testNotificationReceivedMapper() {
         assertNull(bffNotificationsResponseV1Null);
     }
 
+    @Test
```
Comment (AndreaCimini90) [inline src/test/java/it/pagopa/pn/bff/mappers/notifications/NotificationsReceivedMapperTest.java:43]: Move this and test below in a new file

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

**Review by AndreaCimini90 — DISMISSED** [review]

**Review by AndreaCimini90 — APPROVED** [review]

**Pull request merged**